lib/input_data_page.dart

```diff
@@ -0,0 +1,141 @@
+import 'calculated_data_page.dart';
+import 'constants.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'input_data_bloc.dart';
+
+class InputDataPage extends StatefulWidget {
+  @override
+  State<InputDataPage> createState() => _InputDataPageState();
+}
+
+class _InputDataPageState extends State<InputDataPage> {
+  final InputDataBloc _bloc = InputDataBloc(InputDataInitialState());
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Column(
+        children: [
+          const SizedBox(height: 15),
+          BlocBuilder<InputDataBloc, InputDataStates>(
+              bloc: _bloc,
+              builder: (context, state) {
+                print('state $state');
+                if (state is InputDataInitialState) {
+                  return Column(children: [
+                    DatePick(
+                        widgetName: Constants.fromDateNameString,
+                        pick: DateTime(Constants.defaultDate),
+                        bloc: _bloc),
+                    const SizedBox(height: 15),
+                    DatePick(
+                        widgetName: Constants.toDateNameString,
+                        pick: state.lastDate,
+                        bloc: _bloc),
+                  ]);
+                } else if (state is PickedDateState) {
+                  return Column(children: [
+                    DatePick(
+                        widgetName: Constants.fromDateNameString,
+                        pick: state.dateOfBirth,
+                        bloc: _bloc),
+                    const SizedBox(height: 15),
+                    DatePick(
+                        widgetName: Constants.toDateNameString,
+                        pick: state.lastDate,
+                        bloc: _bloc),
+                  ]);
+                } else {
+                  return const Text('Unknown error');
+                }
+              }),
+          const SizedBox(height: 15),
+          ElevatedButton(
+            onPressed: () {
+              if (_bloc.dateOfBirth != DateTime(0000)) {
+                _bloc.calculateAge();
+                Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                        builder: (_) => BlocProvider.value(
+                              value: _bloc,
+                              child: CalculatedDataPage(),
+                            )));
+              } else {
+                ScaffoldMessenger.of(context).showSnackBar(
+                  SnackBar(
+                    content: const Text('Fill in the From date field!'),
+                  ),
+                );
+              }
+            },
+            child: const Text('Calculate Age'),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class DatePick extends StatelessWidget {
```

Comment:
Extract to separate file

Reply:
extracted DatePick class to date_pick_widget.dart file